### ISSUE TITLE
Select all orders selecting orders which are not visible in order grid. 

### ISSUE THREAD

**Issue by shachiarkatkar** [magento/magento2#18983]
### Preconditions (*)
1. Magento 2.2.5
2. Php 7.0

### Steps to reproduce (*)
1. Go to order grid
2. Select all orders on that page
3. Place an order 
4. Without  refreshing page move all orders to hold or any other status
5. Check updated order status for new order

### Expected result (*)
1. Only the order selected should be updated

### Actual result (*)
1. All orders including those which are not visible in order grid are moved to new status.
Order grid before updating status
![image](https://user-images.githubusercontent.com/2101412/47836790-90ac9b80-ddcf-11e8-958e-56fd4c9c957c.png)
Order grid after putting order on hold
![image](https://user-images.githubusercontent.com/2101412/47836808-a91cb600-ddcf-11e8-9fc5-1531282a6d71.png)

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@shachiarkatkar do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by shachiarkatkar** [comment]
@magento-engcom-team give me 2.2.5 instance

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your request. I'm working on Magento 2.2.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-2-5/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-2-5//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-2-5//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by shachiarkatkar** [comment]
@magento-engcom-team the error replicated on vanilla instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @shachiarkatkar thank you for you report, this issue has already fixed in 2.2-develop and 2.3-develop
![selection_188](https://user-images.githubusercontent.com/40756439/47838726-e9bd0500-ddb8-11e8-8ea3-9c6ebcd6bf27.png)
 branch

**Comment by shachiarkatkar** [comment]
@engcom-backlog-nazar can you give commit number here please

**Comment by ghost** [comment]
@shachiarkatkar might be this commit -> https://github.com/magento/magento2/pull/18377/files

**Comment by ghost** [comment]
@shachiarkatkar https://github.com/magento/magento2/pull/14629

**Comment by shachiarkatkar** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-3-develop/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-3-develop//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18983-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by shachiarkatkar** [comment]
@magento-engcom-team 

I am able to replicate the issue on the Magento instance you have given.

I had selected 4 orders to change the status, but during that time another order is placed via frontend or backend

. But this new order is not there in the grid list.

I put those orders to hold, but the 5th order is also moved to hold but that order is not there in my selected orders.

**Comment by shachiarkatkar** [comment]
Hi can anyone help me with this issue

**Comment by shachiarkatkar** [comment]
Please reopen this issue as it is not fixed in 2.3

**Comment by ghost** [comment]
Hi @shachiarkatkar strange but in my screen on 2.3-develop the newly order have not status on hold, however i recheck it.

**Comment by ghost** [comment]
@shachiarkatkar i'm able reproduce if i select all orders, you describe that all orders on page, this issue reproduce only if you select ALL orders.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96239`, `MAGETWO-96240` were created

**Comment by magento-engcom-team** [comment]
Hi @progreg. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by shachiarkatkar** [comment]
@magento-engcom-team give me 2.2.5 instance

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your request. I'm working on Magento 2.2.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar, here is your [Magento instance](http://34.228.235.121/i-18983-2-2-5/).
Admin access: [http://34.228.235.121/i-18983-2-2-5//admin](http://34.228.235.121/i-18983-2-2-5//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by shachiarkatkar** [comment]
Applied the patch but facing same error.

**Comment by shachiarkatkar** [comment]
Correction the patch worked for me thanks @progreg
